### ISSUE TITLE
Doc comment reference for mixin property not recognized

### ISSUE THREAD

**Issue by zoechi** [dart-lang/sdk#31898]
![image](https://user-images.githubusercontent.com/405837/34943650-c947b428-f9fc-11e7-9a25-0204ee0ac6a8.png)

with 
```dart
abstract class EntityMixin<T extends EntityMixin<T>> {
  Sys get sys;
}
```

**Comment by bwilkerson** [comment]
Yeah. So the last time I checked the documentation, doc comments follow markdown _except_ that some identifiers inside square brackets will sometimes be interpreted as references to names defined in Dart code and treated like a magic reference. If there is a single identifier then it is looked up in the scope of the element being documented. If there is a qualified identifier (`id.id`) then the first identifier is looked up in scope and is expected to be a class and the second is expected to be a member of the class. (I believe it also accounts for prefixes, but I'd have to double check.)

There is, unfortunately, some ambiguity because markdown allows single identifier references whose links are provided later in the doc. We don't deal well with that, and there's an open issue for it.

All that to say, I think this is working as originally intended. (Certainly as it was intended the last time I looked at it.)

That said, I think someone needs to think seriously about what we do and don't want to allow in documentation comments and how we handle these ambiguities so that we have a consistent and useful tool for documenting code. Personally, I don't have a problem with extending it to support the use case above, but it would be hard for me to argue that we should support that use case unless we fully generalize it to support _any_ expression. That kind of generalization will take time (both implementation time and execution time) and we have to weigh the benefits against the cost.

(And just to muddy the waters a little more: should we also try to resolve code within code delimiters?)
